giles-eco-tardis/src/main/java/edu/asu/diging/tardis/core/service/impl/ImageExtractionManager.java

```diff
@@ -73,110 +90,86 @@ public void init() {
+        List<edu.asu.diging.gilesecosystem.requests.impl.Page> pages = new ArrayList<>();
+        edu.asu.diging.gilesecosystem.requests.impl.Page requestPage = new edu.asu.diging.gilesecosystem.requests.impl.Page();
+        requestPage.setPageElements(new ArrayList<>());
+        requestPage.setPageNr(request.getPageNr());
+        if (files != null && files.length > 0) {
+            for (File file : files) {
+                PageElement pageElem = new PageElement();
+                pageElem.setContentType("image/png");
+                pageElem.setFilename(file.getName());
+                pageElem.setType("IMAGE");
+                pageElem.setDownloadUrl(
+                restEndpoint + DownloadFileController.GET_FILE_URL
+                    .replace(DownloadFileController.USER_NAME_PLACEHOLDER, request.getUsername())
+                    .replace(DownloadFileController.UPLOAD_ID_PLACEHOLDER, request.getUploadId())
+                    .replace(DownloadFileController.DOCUMENT_ID_PLACEHOLDER, request.getDocumentId())
+                    .replace(DownloadFileController.PAGE_NR, String.valueOf(request.getPageNr()))
+                    .replace(DownloadFileController.FILENAME_PLACEHOLDER, file.getName()));
+                pageElem.setStatus(PageStatus.COMPLETE);
+                requestPage.getPageElements().add(pageElem);
+            }
+            pages.add(requestPage);
+            progressManager.setPhase(ProgressPhase.WIND_DOWN);
+            ICompletionNotificationRequest completedRequest = null;
+            try {
+              completedRequest = requestFactory.createRequest(request.getRequestId(), request.getUploadId());
+            } catch (InstantiationException | IllegalAccessException e) {
+                messageHandler.handleMessage("Could not create request.", e, MessageType.ERROR);
+            }
+            completedRequest.setDocumentId(request.getDocumentId());
+            completedRequest.setFileId(request.getFileId());
+            completedRequest.setNotifier(propertiesManager.getProperty(Properties.NOTIFIER_ID));
+            completedRequest.setStatus(status);
+            completedRequest.setExtractionDate(OffsetDateTime.now(ZoneId.of("UTC")).toString());
+            completedRequest.setPages(pages);
+            completedRequest.setDerivedFile(true);
+            completedRequest.setContentType("image/png");
+            progressManager.setPhase(ProgressPhase.DONE);
+            try {
+                requestProducer.sendRequest(completedRequest,
+                propertiesManager.getProperty(Properties.KAFKA_TOPIC_COMPLETION_NOTIFICATIION));
+            } catch (MessageCreationException e) {
+                messageHandler.handleMessage("Could not send message.", e, MessageType.ERROR);
+            }
+            progressManager.reset();
+         } else {
+             fileService.deleteFile(request.getUsername(), request.getUploadId(), request.getDocumentId(), request.getPageNr(), null);
```

Comment:
I would turn around the if/else logic so you can just say "if there are no files, delete ... (what is it actually deleting, there should be a comment). And then return and loose the else.